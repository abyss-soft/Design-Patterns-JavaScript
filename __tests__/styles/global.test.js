import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { themeLight } from '../../src/styles/themes/theme.light';
import { themeDark } from '../../src/styles/themes/theme.dark';
import GlobalStyle from '../../src/styles/global';

describe('Global style', () => {
  it('renders with a LIGHT theme', () => {
    renderer.create(<GlobalStyle theme={themeLight} />);
    expect(document.head).toMatchSnapshot();
  });

  it('renders with a DARK theme', () => {
    renderer.create(<GlobalStyle theme={themeDark} />);
    expect(document.head).toMatchSnapshot();
  });
});
