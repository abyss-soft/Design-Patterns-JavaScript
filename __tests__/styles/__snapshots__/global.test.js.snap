// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global style renders with a DARK theme 1`] = `
<head>
  <style
    data-styled=""
    data-styled-version="4.3.1"
  >
    
    
/* sc-component-id: sc-global-1470103548 */
body{background:#222222;font-family:'Karla',sans-serif;margin:0;padding:0;} #root{-webkit-box-pack:start;-webkit-justify-content:start;-ms-flex-pack:start;justify-content:start;max-width:640px;margin:auto;padding:0 1rem 3rem;} @media (min-width:769px){#root{-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;padding:0 0 3rem;}} .fixed{height:375px;}
  </style>
</head>
`;

exports[`Global style renders with a LIGHT theme 1`] = `
<head>
  <style
    data-styled=""
    data-styled-version="4.3.1"
  >
    
    
/* sc-component-id: sc-global-1470103548 */
body{background:#FFFFFF;font-family:'Karla',sans-serif;margin:0;padding:0;} #root{-webkit-box-pack:start;-webkit-justify-content:start;-ms-flex-pack:start;justify-content:start;max-width:640px;margin:auto;padding:0 1rem 3rem;} @media (min-width:769px){#root{-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;padding:0 0 3rem;}} .fixed{height:375px;}
  </style>
</head>
`;
